virtcontainers/device/drivers/vfio.go

```diff
@@ -97,17 +114,38 @@ func (device *VFIODevice) DeviceType() config.DeviceType {
 	return device.DevType
 }
 
-// getBDF returns the BDF of pci device
-// Expected input strng format is [<domain>]:[<bus>][<slot>].[<func>] eg. 0000:02:10.0
-func getBDF(deviceSysStr string) (string, error) {
-	tokens := strings.Split(deviceSysStr, ":")
+type VFIODevice_Type int32
+
+const (
+        // 0 is reserved for errors.
+        VFIODevice_TYPE_VFIO_NORMAL  VFIODevice_Type = 1
+        VFIODevice_TYPE_VFIO_MDEV    VFIODevice_Type = 2
+)
 
-	if len(tokens) != 3 {
-		return "", fmt.Errorf("Incorrect number of tokens found while parsing bdf for device : %s", deviceSysStr)
+func getVFIODeviceType(deviceSysStr string) (VFIODevice_Type) {
+
+	tokens := strings.Split(deviceSysStr, ":")
+	if len(tokens) == 3 {
+		return VFIODevice_TYPE_VFIO_NORMAL
+	} else {
+		tokens = strings.Split(deviceSysStr, "-")
+		if len(tokens) == 5 {
+			return VFIODevice_TYPE_VFIO_MDEV
+		}
 	}
+	return 0;
+}
+
+// getBDF returns the BDF of pci device
+// Expected input strng format is [<domain>]:[<bus>][<slot>].[<func>] eg. 0000:02:10.0
+func getBDF(deviceSysStr string) (string) {
```

Comment:
What (should) happen if `getBDF()` and `getSysfsDev()` are given empty strings? It would be great if you could add some unit tests for these functions, short though they are.